app/src/ui/app.tsx

```diff
@@ -1011,11 +1011,18 @@ export class App extends React.Component<IAppProps, IAppState> {
     }
   }
 
-  private onConfirmRepoRemoval = (
+  private onConfirmRepoRemoval = async (
     repository: Repository,
     deleteRepoFromDisk: boolean
   ) => {
-    this.props.dispatcher.removeRepositories([repository], deleteRepoFromDisk)
+    try {
+      await this.props.dispatcher.removeRepositories(
+        [repository],
+        deleteRepoFromDisk
+      )
+    } catch (error) {
+      this.props.dispatcher.presentError(error)
```

Comment:
Is this the right way to ensure that the error is shown?

Reply:
I think you want the `postError` method instead. As per the `presentError` documentation:

https://github.com/desktop/desktop/blob/c080afb5777faa4c2d2a898023ddb56872955067/app/src/ui/dispatcher/dispatcher.ts#L555-L560